common/WhirlyGlobeLib/src/VectorObject.cpp

```diff
@@ -535,116 +590,126 @@ bool VectorObject::pointNearLinear(const Point2d &coord,float maxDistance,const
         return false;
     }
 
+    const Point2f pf = p.cast<float>();
+    const GeoCoord geoCoord(coord.x(),coord.y());
+    GeoMbr searchMbr(geoCoord, geoCoord);
+
     const double maxDistSq = (double)maxDistance * maxDistance;
     for (const auto &shape : shapes)
     {
         if (const auto linear = dynamic_cast<VectorLinear*>(shape.get()))
         {
             const GeoMbr geoMbr = linear->calcGeoMbr();
-            if(geoMbr.inside(GeoCoord(coord.x(),coord.y())))
+            if (!checkBounds(pf, geoCoord, geoMbr, searchMbr, mapView, globeView,
+                             coordAdapter, modelMatFull, frameSize, maxDistance))
```

Comment:
Is `modelMatFull` the right matrix to be using with `pointOn...FromScreen`?

Reply:
That one should have the perspective projection in it.  So yes, that should be right for the flat map.
For the globe things get a little weirder because of the sphere.

Reply:
Looking at other places where those methods are used, it seems like the complications are winding up on the "back" of the globe and in this case I don't really care about that, and the possibility of multiple transforms, which already isn't handled there and so I'm at least not making it worse.

Is there more that needs to be done here?

Reply:
Probably true on the globe issue.
Multiple transforms is for the case in the 2D map where we're zoomed really far out and want to render the world multiple times.  It's been broken since v3.
I'm not sure you need to do that here.  It will probably just work out.